src/Cli/ConfigGenerator.cs

```diff
@@ -1646,24 +1647,137 @@ public static bool TryUpdateExistingEntity(UpdateOptions options, RuntimeConfig
+                    return false;
+                }
+
+                if (hasMappings)
+                {
+                    _logger.LogError("Entity cannot define both 'fields' and 'mappings'. Please use only one.");
+                    return false;
+                }
+
+                if (hasKeyFields)
                 {
+                    _logger.LogError("Entity cannot define both 'fields' and 'key-fields'. Please use only one.");
                     return false;
                 }
+
+                // Merge updated fields with existing fields
+                List<FieldMetadata> existingFields = entity.Fields?.ToList() ?? [];
+                List<FieldMetadata> updatedFieldsList = ComposeFieldsFromOptions(options);
+                Dictionary<string, FieldMetadata> updatedFieldsDict = updatedFieldsList.ToDictionary(f => f.Name, f => f);
+                List<FieldMetadata> mergedFields = [];
+
+                foreach (FieldMetadata field in existingFields)
+                {
+                    if (updatedFieldsDict.TryGetValue(field.Name, out FieldMetadata? updatedField))
+                    {
+                        mergedFields.Add(new FieldMetadata
+                        {
+                            Name = updatedField.Name,
+                            Alias = updatedField.Alias ?? field.Alias,
+                            Description = updatedField.Description ?? field.Description,
+                            PrimaryKey = updatedField.PrimaryKey || field.PrimaryKey
+                        });
+                        updatedFieldsDict.Remove(field.Name); // Remove so only new fields remain
+                    }
+                    else
+                    {
+                        mergedFields.Add(field); // Keep existing field
+                    }
+                }
+
+                // Add any new fields that didn't exist before
+                mergedFields.AddRange(updatedFieldsDict.Values);
+
+                fields = mergedFields;
+
+                // Remove legacy props if fields present
+                updatedSource = updatedSource with { KeyFields = null };
+                updatedMappings = null;
+            }
+            else if (hasMappings || hasKeyFields)
+            {
+                // If mappings or key-fields are provided, convert them to fields and remove legacy props
+                fields = [];
+
+                // Legacy path: allow mappings and key-fields
+                if (hasMappings)
+                {
+                    if (options.Map is null || !TryParseMappingDictionary(options.Map, out updatedMappings))
+                    {
+                        _logger.LogError("Failed to parse mappings from --map option.");
+                        return false;
+                    }
+
+                    foreach (KeyValuePair<string, string> mapping in updatedMappings)
+                    {
+                        fields.Add(new FieldMetadata
```

Comment:
What if there were `existingFields`? We need to merge them with the fields created using the mappings.